src/core/scene/a-scene.js

```diff
@@ -270,13 +270,15 @@ class AScene extends AEntity {
    * @param {bool?} useAR - if true, try immersive-ar mode
    * @returns {Promise}
    */
-  enterVR (useAR) {
+  enterVR (useAR, useOfferSession = false) {
     var self = this;
     var vrDisplay;
     var vrManager = self.renderer.xr;
     var xrInit;
 
     // Don't enter VR if already in VR.
+    if (navigator.xr.offerSession === undefined) { return Promise.resolve('OfferSession is not supported.'); }
+    if (this.is('has-offer-session')) { return Promise.resolve('Already has offerSession.'); }
```

Comment:
Probably don't need this branch here line below should catch if we're already in immersive mode. I think we might not need the `has-offer-session` state

https://github.com/aframevr/aframe/pull/5410/files#diff-ea7d53491f815a744ea89a9a512d67374cb7b8741284b7ed041fde95e53610e1L280

Reply:
`toggleEnterVRButtonIfNeeded` is called multiple times when starting up. The previous pending promise then becomes failed which is flagged as an runtime error by aframe.

Reply:
toggleEnterVRButtonIfNeeded shouldn't be called multiple times. We're doing something wrong I think :)